fwupd: 1.8.3 → 1.8.4

jtojnar opened the pull request:
###### Description of changes

https://github.com/fwupd/fwupd/releases/tag/1.8.4
https://blogs.gnome.org/hughsie/2022/08/30/new-fwupd-1-8-4-release/

Required for https://github.com/NixOS/nixpkgs/pull/182618

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

jtojnar committed:
fwupd: 1.8.3 → 1.8.4

https://github.com/fwupd/fwupd/releases/tag/1.8.4
https://blogs.gnome.org/hughsie/2022/08/30/new-fwupd-1-8-4-release/

jtojnar commented:
Looks like the installed tests fail now even on `HEAD^` but it worked in https://github.com/NixOS/nixpkgs/pull/183686.

superherointj commented:
@ofborg eval

superherointj commented:
Result of `nixpkgs-review pr 189056` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.discover</li>
    <li>libsForQt514.discover</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>fwupd</li>
    <li>gnome-firmware</li>
    <li>gnome.gnome-software</li>
    <li>libsForQt5.discover (libsForQt515.discover ,plasma5Packages.discover)</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.switchboard-plug-about</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-with-indicators</li>
  </ul>
</details>

SuperSandro2000 commented:
Any idea why the tests fail? We should fix them.

jtojnar commented:
Bisected the failure to f7637100655127dffbe14011b45be6a71f0e203d.

jtojnar committed:
nixos/fwupd: add polkit dependency

We are building fwupd daemon with polkit support which means
polkit daemon is required.

Previously polkit was enabled by default via udisks2 but that
stopped with https://github.com/NixOS/nixpkgs/commit/f7637100655127dffbe14011b45be6a71f0e203d
breaking the fwupd installed tests as a result.

Let’s add the polkit dependency to the fwupd module to ensure polkit is available.
